Water-Physics/Assets/Code/Scripts/Environment/WaterFloat.cs

```diff
@@ -0,0 +1,133 @@
+﻿using UnityEngine;
+
+namespace Code.Scripts.Environment
+{
+    [RequireComponent(typeof(Rigidbody))]
+    public class WaterFloat : MonoBehaviour
+    {
+        //public properties
+        public float airDrag = 1;
+        public float waterDrag = 10;
+        public bool affectDirection = true;
+        public bool attachToSurface;
+        public Transform[] floatPoints;
+
+        //used components
+        private Rigidbody rigidbody;
+        private Waves waves;
+
+        //water line
+        private float waterLine;
+        private Vector3[] waterLinePoints;
+
+        //help Vectors
+        private Vector3 smoothVectorRotation;
+        private Vector3 targetUp;
+        private Vector3 centerOffset;
+
+        private Vector3 Center => transform.position + centerOffset;
```

Comment:
Volgens de style guide gebruiken we de `{ get; set;}` syntax i.p.v. dat we er een expression van maken.
```suggestion
        private Vector3 Center
        {
            get { return transform.position + centerOffset; }
        }
```